File: ex-giturb/python/models/basket_of_urbs.py
from pydantic import BaseModel
from typing import List
from models.urb import Urb


class BasketOfUrbs(BaseModel):
    id: int
    urbs: List[Urb] = []

    def is_ready_to_send(self):
        # more than 0, less than 10 but not 7. Should throw exceptions if they do not meet the criteria
        if len(self.urbs) == 0:
            return False
        elif len(self.urbs) == 7:
            raise Exception("Basket cannot have 7 Urbs")
        elif len(self.urbs) >= 10:
            raise Exception("Basket cannot have more than 10 urbs")
        else:
            return True

File: ex-giturb/dotnet/giturb/Models/BasketOfUrbs.cs
public class BasketOfUrbs 
{
    public required int Id { get; set; }
    public List<Urb> Urbs { get; set; } = new();

    public bool IsReadyToSend()
    {
        if (Urbs.Count == 0)
        {
            return false;
        }
        if (Urbs.Count == 7)
        {
            throw new Exception("Basket cannot have 7 Urbs");
        }
        if (Urbs.Count >= 10)
        {
            throw new Exception("Basket cannot have more than 10 urbs");
        }
        return true;
    }
}
